modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,20 +370,107 @@ public void testGetSuccess(String id, String attributes,
+        };
+    }
+
+    @Test(dataProvider = "dataForGetUserNotFoundException")
+    public void testGetUserNotFoundException(String id, String attributes, String excludeAttributes)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        abstractResourceManager.when(() ->
+                AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+        Mockito.when(userManager.getUser(id, requiredAttributes)).thenReturn(null);
+
+        SCIMResponse scimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND);
+    }
+
+    @DataProvider(name = "dataForGetCharonException")
+    public Object[][] dataToGetCharonException() {
+
+        return new Object[][]{
+                {"1234", "userName", null, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetCharonException")
+    public void testGetUserCharonException(String id, String attributes, String excludeAttributes,
+                                           int expectedScimResponseStatus)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+        Mockito.when(userManager.getUser(id, requiredAttributes))
+                .thenThrow(CharonException.class);
+
+        SCIMResponse scimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForGetBadRequestException")
+    public Object[][] dataToGetCharonBadRequestException() {
```

Comment:
this data provider is only for BadRequestException right? remove the `Charon` part.

Reply:
Fixed in commit 6f5ca461fa07ac4b49182320325156a58b80b910